### ISSUE TITLE
TypeError: Object of type datetime is not JSON serializable

### ISSUE THREAD

**Issue by WhyRandomUser** [arcsight-unofficial/arcsight-logger-api-sdk#2]
Hi,

When ever I attempt to drill down or set some form of time parameter I keep getting the following error:
TypeError: Object of type datetime is not JSON serializable
I'm currently using python3. @P1llus 

Thanks

**Comment by P1llus** [comment]
Thanks for the feedback, will look into this tomorrow or in the weekend!
Could you be able to provide a small example on how you implemented it?

**Comment by P1llus** [comment]
Hey @WhyRandomUser .
After a quick thought, based on your error message I think I know what is wrong.

When sending an API call to the Logger using the SDK, all requests are converted into JSON before sending it off in the HTTP request.
As JSON does not work on objects itself, you have to ensure that all data you pass into it is either a String, Integer, Boolean, Array etc.

TLDR(Explanation ended up being longer than expected):
You need to convert the datetime object to string before passing it into one of the functions.

Let me show you an example:

```python
# Importing the library for testing
from datetime import datetime

# Get the current time, this is stored as an Object, and not a string.
CURRENT_TIME = datetime.now

# Set the start time to NOW minus 40 minutes
DRILLDOWN_START = CURRENT_TIME - timedelta(minutes=40)

# Set the endtime to NOW
DRILLDOWN_END = CURRENT_TIME - timedelta(minutes=10)

# Trying to run a drilldown
loggersdk.drilldown(HOST, authtoken, SEARCH_ID, DRILLDOWN_START, DRILLDOWN_END)
```
This will fail, as the TYPE of DRILLDOWN_START and DRILLDOWN_END is still a datetime object.

In most cases, you either need to use a function on datetime that converts the object to string, called strftime [referenced here:](https://docs.python.org/3/library/datetime.html?highlight=strftime#strftime-and-strptime-behavior) or anything similar.

I personally hate managing times and dates in python, it is actually easy, but since I want to have as little third party dependencies as possible, I can make it hard for myself.

Personally I would recommend using other more straight forward date/time libraries in your script, and just pass the correct time format. The format it expects is `TIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%f%z'`

I also created an example documentation one how to do a basic implementation of each functionality, located [here](https://github.com/arcsight-unofficial/arcsight-logger-api-sdkexample/blob/master/client.py).

If you look at my date and time implementation, it looks something like this, it is ugly I know, but I wanted the example to support any timezone and be based on steps instead of one long confusing variable, without third party libraries, the flow goes something like this:

1. Set format so i can change format of all dates at one place
2. Get the current offset from UTC in your timezone.
3. Get current time with your timezone.
4. Set the Drilldown Start, The reason we do it twice is because it has to be a Datetime Object to be able to do operations on it (like plus/minus).
5. Convert the format to the one ArcSight recognizes.
6. Same for Drilldown end.

```python
import time
import json
from datetime import datetime
from datetime import timedelta
import loggersdk

TIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%f%z'
CURRENT_TIMEZONE = datetime.utcnow().astimezone().tzinfo
CURRENT_TIME = datetime.now(tz=CURRENT_TIMEZONE)
DRILLDOWN_START = CURRENT_TIME - timedelta(minutes=40)
DRILLDOWN_START = DRILLDOWN_START.isoformat(timespec='milliseconds')
DRILLDOWN_END = CURRENT_TIME - timedelta(minutes=10)
DRILLDOWN_END = DRILLDOWN_END.isoformat(timespec='milliseconds')
```

Hope this helps! :) I keep the issue open a bit longer, and close it if there is no new updates! Thanks for testing out the SDK and taking your time to provide feedback.

**Comment by WhyRandomUser** [comment]
Hi,

It seems my code was missing just this and it now works beautifully. Would you happen to know however why there seems to be a limit in the results in queries? I seem to always be caped at 1000.

Thanks

**Comment by P1llus** [comment]
The default is indeed 1000 entries, and the max is 10k, which is where offsets comes in, this is also in the example and documentation I made :)

If you clone this repo and just open the index.html you should be good to go: https://github.com/arcsight-unofficial/arcsight-logger-api-documentation

When you try to retrieve an event either through `events` or `chart_data` it is possible to specify length as one of the optional parameters.
Only the required parameters are actually needed to use the SDK, but the optional parameters in the documentation can always be added as a **kwarg.

Example
```
kwargs = {
    'length': 30,
    'offset': 2
}

chart_data = loggersdk.chart_data(HOST, authtoken, SEARCH_ID, **kwargs)
```

If your search returns 50 000 events, and you want to retrieve them all, you need to get the maximum 10 000 length, then set the offset to 10 001 and run it again:

```
kwargs = {
    'length': 10000,
    'offset': 0
}

event_data = loggersdk.events(HOST, authtoken, SEARCH_ID, **kwargs)

kwargs = {
    'length': 10000,
    'offset': 10001
}

event_data = loggersdk.events(HOST, authtoken, SEARCH_ID, **kwargs)
```

Of course there is much prettier ways to implement it, but this seems the most straight forward.
